networking.md

```diff
@@ -0,0 +1,92 @@
+# Filecoin Network
+
+The filecoin network is built using libp2p building blocks, as transports and protocols, as well as some additional filecoin specific protocols as outlined in [Network Protocols](./network-protocols.md).
+
+## Required Protocols
+
+Every full node must support the following libp2p protocols:
+
+ - [gossipsub](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub) ([for data announcements](./data-propagation.md))
+ - [bitswap] ([for data exchange](./))
+ - Filecoin specific Protocols:
+   - Hello Handshake
+   - StorageDeal
+   - BlockSync
+
+## Transports, Streams & Encryption
+
+Connections between nodes are end-to-end-encrypted and -authenticated, thus every node has a `peerId` associated with it. Any node can also support unencrypted or unauthenticated connection request, but it is not required to do so.
+
+Filecoin protocols are run as multiple streams over a single connection. Only one connection should exist between two nodes at any time, multiple streams within that connections must be used to exchange different protocols.
+
+This document doesn't make any assumptions on which transports and multiplexing features a particular node offers, however it must at least be able to respond and communicate through ABC AND XYZ
+
+{{% notice todo %}}
+Define minimum requirements in terms of network/e2ee-protocols and multiplexing mechanism. Should we directly jump to yamux?
```

Comment:
We should go with secio and mplex for now, if tls 1.3 becomes mature enough all around, we could swap it for secio, but we're not there yet. Mplex also tends to be more performant, and drastically easier to implement than yamux (downside being that it doesnt have great backpressure).